### PR TITLE
📝 Contribution documents

### PR DESCRIPTION
**Pull request by C-Valen**
## Checklist:

- [X] Linting passed
- [X] Type checks passed
- [X] Tests suite passed
- [x] Commits consolidated
- [ ] Changelog updated (n/a)
- [ ] Test cases added/updated (n/a)
- [ ] Jira ticket updated (n/a)

## Summary:

Provides documents for pull request description template and contribution guidelines.

## Changes:

- Adds `pull_request_template.md` to  `.github` dir
- Adds `contribution.md` to `/docs`

## Considerations:

- Just a draft proposal for both documents, we can use this PR to discuss/adapt the info provided or huddle about it
- I think we should also update [OSIM Developer Guide](https://github.com/RedHatProductSecurity/osim/blob/doc/contribution-docs/docs/overview.md)
- Would be nice to extend [OSIM Developer Guide](https://github.com/RedHatProductSecurity/osim/blob/doc/contribution-docs/docs/overview.md) with some more information like `Ecosystem`, `Useful tools` ...

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# [OSIDB-ID] [Title]
+
+## Checklist:
+
+- [ ] Linting passed
+- [ ] Type checks passed
+- [ ] Tests suite passed
```
Comment (JakubFrejlach) [inline .github/pull_request_template.md:7]: I am not sure about these 3 points, they seem kinda redundant to me as they are all checked via github workflow. Also making some additional changes to the PR might cause any of these to fail again and thus rendering this checkbox outdated.

Idea: On the other hand, we currently only run one github workflow which just runs the deploy script which does all the linting, type check, etc. It is sometimes kinda hard to find from one long log file what exactly failed.

Comment (C-Valen) [reply .github/pull_request_template.md:7]: > I am not sure about these 3 points, they seem kinda redundant to me as they are all checked via github workflow. Also making some additional changes to the PR might cause any of these to fail again and thus rendering this checkbox outdated.

That's a good consideration, being true that GH workflow performs automatic checks on those I think if we will have a PR checklist they should be considered as it kind of create the habit of checking those points before pushing/creating PR.

As I see it GH workflow is to ensure code is correct, and checklist points to make the developer conscious on the common actions before opening a PR.

> Idea: On the other hand, we currently only run one github workflow which just runs the deploy script which does all the linting, type check, etc. It is sometimes kinda hard to find from one long log file what exactly failed.

So you suggest to have an independent workflow script with only required steps for PRs? That'd be a a good idea as well.

Comment (JakubFrejlach) [reply .github/pull_request_template.md:7]: For example in OSIDB we have a github workflow setup like this:
image

Comment (superbuggy) [reply .github/pull_request_template.md:7]: I like this in theory, I have to think through how this might change our build process

**Review by d10n — APPROVED** [review]

File: .github/pull_request_template.md (hunk 1 above)
Comment (d10n) [reply .github/pull_request_template.md:7]: For now, the checklist doesn't do any harm, but we can remove items that get automated with PR push actions

**Pull request merged**